clang/include/clang/CIRFrontendAction/CIRGenAction.h

```diff
@@ -128,6 +129,59 @@ class EmitObjAction : public CIRGenAction {
   EmitObjAction(mlir::MLIRContext *mlirCtx = nullptr);
 };
 
+class AnalysisOnlyActionBase : public clang::CodeGenAction {
```

Comment:
Add a comment here: `// Used for -fclangir-analysis-only: use CIR analysis but still use original LLVM codegen path`

Reply:
Done